packages/test-interface-criteria/src/criteria/criteriaUtils.js

```diff
@@ -142,86 +181,50 @@ function createCriteria({
-      //   if (thisTestDataRequirementValue === null || baseTestDataRequirementValue === null) {
-      //     return thisTestDataRequirementValue ?? baseTestDataRequirementValue;
-      //   }
-      //   return chooseThisRequirementOverThatRequirement(thisTestDataRequirementValue, baseTestDataRequirementValue)
-      //     ? thisTestDataRequirementValue
-      //     : baseTestDataRequirementValue;
-      // };
-      // return {
-      //   ...baseTestDataShape,
-      //   ...testDataShapeFactory,
-      //   startDateMin: chooseNarrowerRequirement('startDateMin', (thisValue, thatValue) => moment(thisValue).isBefore(thatValue)),
-      //   startDateMax: chooseNarrowerRequirement('startDateMax', (thisValue, thatValue) => moment(thisValue).isAfter(thatValue)),
-      //   validFromMin: chooseNarrowerRequirement('validFromMin', (thisValue, thatValue) => moment(thisValue).isBefore(thatValue)),
-      //   validFromMax: chooseNarrowerRequirement('validFromMax', (thisValue, thatValue) => moment(thisValue).isAfter(thatValue)),
-      //   // durationMin: chooseNarrowerRequirement('durationMin',
-      //   //   (thisValue, thatValue) => moment.duration(thisValue).asMilliseconds() > moment.duration(thatValue).asMilliseconds()),
-      //   // durationMax: chooseNarrowerRequirement('durationMax',
-      //   //   (thisValue, thatValue) => moment.duration(thisValue).asMilliseconds() < moment.duration(thatValue).asMilliseconds()),
-      //   remainingCapacityMin: chooseNarrowerRequirement('remainingCapacityMin',
-      //     (thisValue, thatValue) => thisValue > thatValue),
-      //   remainingCapacityMax: chooseNarrowerRequirement('remainingCapacityMax',
-      //     (thisValue, thatValue) => thisValue < thatValue),
-      //   // TODO There are requirements that have no clear way of merging (e.g. eventStatusIsEventScheduled).
-      //   // These should error if there is an overwrite
-      //   //
-      //   // [obsolete] TODO eventStatusOptions cannot be merged in this way. Either:
-      //   // - MAKE IT MERGEABLE
-      //   //   - make a more generic method for merging values so that eventStatusOptions can use an intersection
-      //   // - MAKE IT UN-MERGEABLE
-      //   //   - If a base criteria has a value for this and a child criteria has a value, raise an error.
-      //   //   - I think it would also be clearer that it cannot be merged if it was a singular value ("eventStatusOption")
-      //   //     rather than an array
-      // };
+      return extendTestDataShape(baseTestDataShape, thisTestDataShape, name);
     },
   };
 }
 
+/**
+ * Merge constraints so that the result has the simplest representation of the combination of all constraints.
+ *
+ * @param {TestDataShape} baseTestDataShape
+ * @param {TestDataShape} extraTestDataShape
+ * @param {string} criteriaName
+ * @return {TestDataShape}
+ */
+function extendTestDataShape(baseTestDataShape, extraTestDataShape, criteriaName) {
```

Comment:
refactored this into a function so it could be used to add the openBookingFlowRequirement ShExs to criteria ShExs based on the testOpenBookingFlow field